src/Microsoft.DotNet.Interactive.HttpRequest/HttpRequestKernel.cs

```diff
@@ -269,192 +234,61 @@ private static HttpRequestMessage GetRequestMessage(ParsedHttpRequest parsedRequ
         return response;
     }
 
-    private void PublishDiagnostics(KernelInvocationContext context, KernelCommand command, IEnumerable<Diagnostic> diagnostics)
+    private void PublishDiagnostics(KernelInvocationContext context, KernelCommand command, IReadOnlyCollection<Diagnostic> diagnostics)
     {
         if (diagnostics.Any())
         {
             var formattedDiagnostics =
                 diagnostics
                     .Select(d => d.ToString())
                     .Select(text => new FormattedValue(PlainTextFormatter.MimeType, text))
-                    .ToImmutableArray();
+                    .ToArray();
 
             context.Publish(new DiagnosticsProduced(diagnostics, command, formattedDiagnostics));
         }
     }
 
     Task IKernelCommandHandler<RequestDiagnostics>.HandleAsync(RequestDiagnostics command, KernelInvocationContext context)
     {
-        var requestsAndDiagnostics = InterpolateAndGetDiagnostics(command.Code);
-        var diagnostics = requestsAndDiagnostics.SelectMany(r => r.Diagnostics);
+        var parseResult = HttpRequestParser.Parse(command.Code);
+        var requestsAndDiagnostics = GetAllDiagnostics(parseResult);

```

Comment:
nit: The variable name is confusing.

Reply:
This was a refactoring artifact. Inlined.